### PR TITLE
Added audience validation

### PR DESCRIPTION
**Pull request by dev-tim**
Hi! I added logic for audience validation in jwt token. 
Here is one more task to have test audience for tests that we could put in this library.

### CODE REVIEW

File: app/com/yetu/oauth2resource/services/tokenvalidation/OAuth2TokenValidationService.scala [hunk 1]
```diff
@@ -81,21 +82,43 @@ class OAuth2TokenValidationService(oAuth2ProviderSettings: OAuth2ProviderSetting
...
+      } else {
+        Logger.error(s"Token validation failed due to expiration ex time is $jwt")
+        Left(new TokenExpiredException(s"Expired access token with $jwt"))
+      })
+  }
+
+  def verifyAudience(jwt: ValidationResponse) = {
+    def ifNoAudience: Either[Exception, ValidationResponse] = {
+      val ex = InvalidAudienceException("Token is corrupted on expiration data provided")
+      Logger.error("Validation exception, expired data is not provided", ex)
```
Comment (jschaul) [inline app/com/yetu/oauth2resource/services/tokenvalidation/OAuth2TokenValidationService.scala:109]: the error message is about audience problems. The text states "expired data is not provided". Please change the error message.

File: app/com/yetu/oauth2resource/services/tokenvalidation/OAuth2TokenValidationService.scala [hunk 2]
```diff
@@ -81,21 +82,43 @@ class OAuth2TokenValidationService(oAuth2ProviderSettings: OAuth2ProviderSetting
...
+      Left(ex)
+    }
+
+    jwt.aud.fold(ifNoAudience)(aud =>
+      Defaults.targetAudience.intersect(aud.split(" ").toList) match {
```
Comment (jschaul) [inline app/com/yetu/oauth2resource/services/tokenvalidation/OAuth2TokenValidationService.scala:114]: you force a user of this library to use the `Defaults.targetAudience` field, which is inflexible. The audience should be a parameter that can be passed to the validation. Isn't there already an audience validation in the JWT library in-built that you can use directly?

**Comment by jschaul** [comment]
1) Could you add a test confirming that a correct audience is accepted?
2) Could you please only merge pull requests AFTER someone has reviewed them, not before?

File: app/com/yetu/oauth2resource/services/tokenvalidation/OAuth2TokenValidationService.scala (hunk 2 above)
Comment (dev-tim) [inline app/com/yetu/oauth2resource/services/tokenvalidation/OAuth2TokenValidationService.scala:114]: There is a possibility to pass audience list as config parameter. 
Default object represents hash of values, taken from params.

File: app/com/yetu/oauth2resource/services/tokenvalidation/OAuth2TokenValidationService.scala (hunk 1 above)
Comment (dev-tim) [inline app/com/yetu/oauth2resource/services/tokenvalidation/OAuth2TokenValidationService.scala:109]: :+1:

**Comment by dev-tim** [comment]
Test is there - if no encoded to the token, then test will not pass. For this purpose we have right access token but wrong aud in config.

**Pull request merged**